### PR TITLE
Server: Upgrade to CSF3

### PR DESCRIPTION
**Pull request by jonspalmer**
## What I did

Upgraded server to CSF3 format

### CODE REVIEW

**Comment by jonspalmer** [comment]
I couldn't get Default Title generation to work in the example app. I suspect its because we can't use the new stories declaration in main.js as we have to pick out json/yaml files

File: lib/core-common/src/utils/normalize-stories.ts [hunk 1]
```diff
@@ -24,10 +27,17 @@ export const normalizeStoriesEntry = (
...
-      titlePrefix = DEFAULT_TITLE_PREFIX;
       files = DEFAULT_FILES;
+      titlePrefix = DEFAULT_TITLE_PREFIX;
     } else {
-      glob = entry;
+      const match = entry.match(GLOB_REGEX);
+      if (match) {
+        directory = match.groups.directory;
+        files = match.groups.files;
+        titlePrefix = DEFAULT_TITLE_PREFIX;
```
Comment (jonspalmer) [inline lib/core-common/src/utils/normalize-stories.ts:37]: @shilman I added this to support common glob cases. This feels powerful for backwards compatibility vs forcing folks to use the directory/files config but i guess we could do with out it. Thoughts?

Comment (shilman) [reply lib/core-common/src/utils/normalize-stories.ts:37]: @tmeasday WDYT? I didn't add this in the first place due to "too much magic", but it doesn't seem too bad 🤷‍♂️

**Review by shilman — APPROVED** [review]
Great! Cleans a bunch of stuff up 💯

File: lib/core-common/src/utils/normalize-stories.ts (hunk 1 above)
Comment (jonspalmer) [reply lib/core-common/src/utils/normalize-stories.ts:37]: I guess it depends on what we want the end state to be. Without this change folks will have to upgrade to the simple version without files extensions. However, that forces users of server to use the directory/files version vs a regex (there is no way to configure default files)

Comment (tmeasday) [reply lib/core-common/src/utils/normalize-stories.ts:37]: I think this makes sense, although I am not 100% clear on the implications. Does it just mean if you are using the `dir/**/*.stories.js` type syntax you can now leave titles off components where you couldn't before?

That seems like a good thing to me.

**Review by tmeasday — APPROVED** [review]
CSFv3 is great for the server use case 👍

**Review by shilman — CHANGES_REQUESTED** [review]
@jonspalmer Can you please split the CSF3 glob processing change out into a separate PR? I'd like to track it separately from the Server work since it affects all frameworks.

**Comment by jonspalmer** [comment]
Closing in favor of the separated PRs #15697 and #15698

**Pull request closed without merging**